### PR TITLE
Feature/sc 108631/asset tracker build with quectel gnss

### PR DESCRIPTION
**Pull request by mehmood-n**
The Tracker Edge application also supports the Quectel GNSS module. There are Quectel-specific messages handled by this GPS parser along with support for calculating and storing satellite description messages. The changes here are to support this [pull request](https://github.com/particle-iot/asset-tracker-fw-private/pull/115).

### CODE REVIEW

File: src/gps/gps.cpp [hunk 1]
```diff
@@ -56,6 +57,10 @@
 #define CTN(x)              ((x) - '0')
 #define CHTN(x)             (((x) >= '0' && (x) <= '9') ? ((x) - '0') : (((x) >= 'a' && (x) <= 'z') ? ((x) - 'a' + 10) : (((x) >= 'A' && (x) <= 'Z') ? ((x) - 'A' + 10) : 0)))
 
+static uint8_t  gGsvType = 0;
```
Comment (eberseth) [inline src/gps/gps.cpp]: no prefix

File: src/gps/gps.cpp [hunk 2]
```diff
@@ -172,10 +177,29 @@ parse_term(gps_t* gh) {
 #if GPS_CFG_STATEMENT_GPGSA
         } else if (!strncmp(gh->p.term_str, "$GPGSA", 6) || !strncmp(gh->p.term_str, "$GNGSA", 6)) {
             gh->p.stat = STAT_GSA;
+            /* This message is a delimiter to indicate all GSV messages have been received.
+               Reset the counter of satellites in view for the next cycle. */
+            gSivTotal = 0;
 #endif /* GPS_CFG_STATEMENT_GPGSA */
 #if GPS_CFG_STATEMENT_GPGSV
-        } else if (!strncmp(gh->p.term_str, "$GPGSV", 6) || !strncmp(gh->p.term_str, "$GNGSV", 6)) {
+        } else if( !strncmp(gh->p.term_str, "$GPGSV", 6) ) {
```
Comment (eberseth) [inline src/gps/gps.cpp]: There are a lot of strncmp calls.  Ideally we'd want to pattern match to `$G([PNLABQ])GSV`.  Consider this instead:
```cpp
} else if( !strncmp(gh->p.term_str, "$G", 2) && !strncmp(&gh->p.term_str[3], "GSV", 3) ) {
    switch (strncmp(&gh->p.term_str[2])
```

File: src/gps/gps.cpp [hunk 3]
```diff
@@ -257,29 +298,53 @@ parse_term(gps_t* gh) {
...
                 if (gh->p.term_num >= 4 && gh->p.term_num <= 19) {  /* Check current term number */
-                    uint8_t index, term_num = gh->p.term_num - 4;   /* Normalize term number from 4-19 to 0-15 */
+                    uint8_t term_num = gh->p.term_num - 4;   /* Normalize term number from 4-19 to 0-15 */
                     uint16_t value;
+                    static uint16_t idx = 0; /* Keep a running count of satellites for this cycle */
+
+                    /* Check if new set of GSV messages is being parsed. Flag this so
+                       that the parser will accumulate the GSV messages instead
+                       of simply saving the last one received */
+                    if( true == newCycle ) {
```
Comment (eberseth) [inline src/gps/gps.cpp]:
```cpp
if (newCycle) {
```

File: src/gps/gps.cpp [hunk 4]
```diff
@@ -342,9 +407,140 @@ parse_term(gps_t* gh) {
...
+                break;
+        }
+    } else if (gh->p.stat == STAT_QUECTEL_VEH) {
+        static uint8_t msg_type;
+        switch (gh->p.term_num) {
+            case 1:
+                msg_type = (uint8_t)parse_number(gh, NULL);
+                break;
+            case 3:
+                if( 1 == msg_type )
```
Comment (eberseth) [inline src/gps/gps.cpp]: Use another `switch/case`

File: src/gps/gps.cpp [hunk 5]
```diff
@@ -523,7 +719,9 @@ copy_from_tmp_memory(gps_t* gh) {
         gh->altitude = gh->p.data.gga.altitude;
         gh->geo_sep = gh->p.data.gga.geo_sep;
         gh->sats_in_use = gh->p.data.gga.sats_in_use;
+        gh->dop_h = gh->p.data.gga.dop_h;
         gh->fix = gh->p.data.gga.fix;
+        // Log.info("gpsp fix: %u", gh->fix);
```
Comment (eberseth) [inline src/gps/gps.cpp]: delete

File: src/gps/gps.cpp (hunk 1 above)
Comment (mehmood-n) [reply src/gps/gps.cpp]: Will fix

File: src/gps/gps.cpp (hunk 5 above)
Comment (mehmood-n) [reply src/gps/gps.cpp]: Will fix

File: src/gps/gps.cpp (hunk 3 above)
Comment (mehmood-n) [reply src/gps/gps.cpp]: Will fix

File: src/gps/gps.cpp (hunk 4 above)
Comment (mehmood-n) [reply src/gps/gps.cpp]: Will fix

File: src/gps/gps.cpp (hunk 2 above)
Comment (mehmood-n) [reply src/gps/gps.cpp]: Thanks for suggestion. Will implement.

**Review by eberseth — APPROVED** [review]

**Pull request merged**